dask/dataframe/rolling.py

```diff
@@ -74,17 +88,30 @@ def map_overlap(func, df, before, after, *args, **kwargs):
     df_name = df._name
 
     dsk = df.dask.copy()
-    if before:
+    if before and isinstance(before, int):
         dsk.update({(name_a, i): (M.tail, (df_name, i), before)
                     for i in range(df.npartitions - 1)})
         prevs = [None] + [(name_a, i) for i in range(df.npartitions - 1)]
+    elif isinstance(before, datetime.timedelta):
+        # Assumes monotonic (increasing?) index
+        f = lambda prev, current, before: prev[prev.index > (current.index.min() - before)]
```

Comment:
If possible it would be nice to avoid the lambda.  Dynamically generated functions end up being expensive to serialize in a distributed setting.  It is common to make small private functions within the same module that are importable (and so trivially pickleable)